pkg/reconciler/autoscaling/kpa/kpa_test.go

```diff
@@ -181,6 +185,130 @@ func markResourceNotOwned(rType, name string) PodAutoscalerOption {
+	overscale := minScale + 1
+
+	underscaledDeployment := deploy(testNamespace, testRevision, func(d *appsv1.Deployment) {
+		d.Spec.Replicas = ptr.Int32(underscale)
+	})
+	overscaledDeployment := deploy(testNamespace, testRevision, func(d *appsv1.Deployment) {
+		d.Spec.Replicas = ptr.Int32(overscale)
+	})
+
+	minScalePatch := clientgotesting.PatchActionImpl{
+		ActionImpl: clientgotesting.ActionImpl{Namespace: testNamespace},
+		Name:       deployName,
+		Patch:      []byte(fmt.Sprintf(`[{"op":"replace","path":"/spec/replicas","value":%d}]`, minScale)),
+	}
+
+	inactiveKpa := kpa(testNamespace, testRevision, markInactive, withMinScale(int(minScale)), WithPAStatusService(testRevision))
+	activatingKpa := kpa(testNamespace, testRevision, markActivating, withMinScale(int(minScale)), WithPAStatusService(testRevision))
+	activeKpa := kpa(testNamespace, testRevision, markActive, withMinScale(int(minScale)), WithPAStatusService(testRevision))
+
+	defaultSks := sks(testNamespace, testRevision, WithDeployRef(deployName), WithSKSReady)
+	defaultMetricsSvc := metricsSvc(testNamespace, testRevision, withSvcSelector(usualSelector))
+
+	underscaledEndpoints := makeSKSPrivateEndpoints(int(underscale), testNamespace, testRevision)
+	overscaledEndpoints := makeSKSPrivateEndpoints(int(overscale), testNamespace, testRevision)
+
+	table := TableTest{{
+		Name: "underscaled, PA inactive",
+		// No-op
+		Key: key,
+		Objects: []runtime.Object{
+			inactiveKpa, underscaledEndpoints, underscaledDeployment,
+			defaultSks, defaultMetricsSvc,
+		},
+	}, {
+		Name: "underscaled, PA activating",
+		// Scale to `minScale`
+		Key: key,
+		Objects: []runtime.Object{
+			activatingKpa, underscaledEndpoints, underscaledDeployment,
+			defaultSks, defaultMetricsSvc,
+		},
+		WantPatches: []clientgotesting.PatchActionImpl{
+			minScalePatch,
+		},
+	}, {
+		Name: "underscaled, PA active",
+		// Mark PA "activating"
+		Key: key,
+		Objects: []runtime.Object{
+			activeKpa, underscaledEndpoints, underscaledDeployment,
+			defaultSks, defaultMetricsSvc,
+		},
+		WantStatusUpdates: []clientgotesting.UpdateActionImpl{{
+			Object: activatingKpa,
+		}},
+	}, {
+		Name: "overscaled, PA inactive",
+		// No-op
+		Key: key,
+		Objects: []runtime.Object{
+			inactiveKpa, overscaledEndpoints, overscaledDeployment,
+			defaultSks, defaultMetricsSvc,
+		},
+	}, {
+		Name: "overscaled, PA activating",
+		// Scale to `minScale` and mark PA "active"
+		Key: key,
+		Objects: []runtime.Object{
+			activatingKpa, overscaledEndpoints, overscaledDeployment,
+			defaultSks, defaultMetricsSvc,
+		},
+		WantPatches: []clientgotesting.PatchActionImpl{
+			minScalePatch,
+		},
+		WantStatusUpdates: []clientgotesting.UpdateActionImpl{{
+			Object: activeKpa,
+		}},
+	}, {
+		Name: "overscaled, PA active",
+		// No-op
+		Key: key,
+		Objects: []runtime.Object{
+			activeKpa, overscaledEndpoints, overscaledDeployment,
+			defaultSks, defaultMetricsSvc,
+		},
+	}}
+
+	defer logtesting.ClearAll()
```

Comment:
Should we be doing this for each row of the table testing?

Reply:
This should work, I think.

Reply:
Happy to go either way, I am 100% cargo cult-ing from the other tests in this suite.

Reply:
Let's follow up then, don't sweat it.